src/python/pants/backend/python/goals/tailor_test.py

```diff
@@ -132,6 +139,47 @@ def test_find_putative_targets_subset(rule_runner: RuleRunner) -> None:
     )
 
 
+def test_find_putative_targets_for_entry_points(rule_runner: RuleRunner) -> None:
+    mains = ("main1.py", "main2.py", "main3.py")
+    rule_runner.write_files(
+        {
+            f"src/python/foo/{name}": textwrap.dedent(
+                """
+            if __name__ == "__main__":
+                main()
+            """
+            )
+            for name in mains
+        }
+    )
+    rule_runner.add_to_build_file(
+        "src/python/foo",
+        "pex_binary(name='main1', entry_point='main1.py')\n"
+        "pex_binary(name='main2', entry_point='foo.main2')\n",
+    )
```

Comment:
FYI, this should go into the `write_files({}` as `src/python/foo/BUILD`. Not blocking and you can save trees, but I'm trying hard to kill off `.add_to_build_file()` and `.create_file()`.